src/clients/scopes/AddScopeDialog.tsx

```diff
@@ -50,55 +100,207 @@ export const AddScopeDialog = ({
     toggleDialog();
   };
 
+  const onFilterTypeDropdownToggle = () => {
+    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
+  };
+
+  const onProtocolTypeDropdownToggle = () => {
+    setIsProtocolTypeDropdownOpen(!isProtocolTypeDropdownOpen);
+  };
+
+  const onFilterTypeDropdownSelect = (filterType: string) => {
+    if (filterType === "Name") {
+      setFilterType(FilterType.Protocol);
+    }
+    if (filterType === "Protocol") {
+      setFilterType(FilterType.Name);
+    }
+    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
+  };
+
+  const onProtocolTypeDropdownSelect = (protocolType: string) => {
+    if (protocolType === "SAML") {
+      setProtocolType(ProtocolType.SAML);
+    }
+    if (protocolType === "OpenID Connect") {
+      setProtocolType(ProtocolType.OpenIDConnect);
+    }
+    if (protocolType === "All") {
+      setProtocolType(ProtocolType.All);
+    }
+    setIsProtocolTypeDropdownOpen(!isProtocolTypeDropdownOpen);
+  };
+
+  const protocolTypeOptions = [
+    <SelectOption key={1} value={ProtocolType.SAML}>
+      {t("protocolTypes.saml")}
+    </SelectOption>,
+    <SelectOption key={2} value={ProtocolType.OpenIDConnect}>
+      {t("protocolTypes.openIdConnect")}
+    </SelectOption>,
+    <SelectOption key={3} value={ProtocolType.All} isPlaceholder>
+      {t("protocolTypes.all")}
+    </SelectOption>,
+  ];
+
   return (
     <Modal
       variant={ModalVariant.medium}
-      title={t("addClientScopesTo", { clientId: "test" })}
+      title={
+        isClientScopesConditionType
+          ? t("addClientScope")
+          : t("addClientScopesTo", { clientName: clientName })
```

Comment:
This can be shortened by omitting the value.
```suggestion
          : t("addClientScopesTo", { clientName })
```